numpy/core/tests/examples/cython/checks.pyx

```diff
@@ -77,3 +77,88 @@ def make_iso_8601_datetime(dt: "datetime"):
         cnp.NPY_NO_CASTING,
     )
     return result
+
+
+cdef cnp.NpyIter* npyiter_from_npyiter_obj(object it):
+    # this function supports up to 3 iterator operands
+    cdef:
+        cnp.NpyIter* cit
+        # cnp.PyArray_Descr* op_dtypes[5]
+        cnp.npy_uint32 op_flags[3]
+        cnp.PyArrayObject* ops[3]
+        cnp.npy_uint32 flags = 0
+
+    if it.has_index:
+        flags |= cnp.NPY_ITER_C_INDEX
+    if it.has_delayed_bufalloc:
+        flags |= cnp.NPY_ITER_BUFFERED | cnp.NPY_ITER_DELAY_BUFALLOC
+    if it.has_multi_index:
+        flags |= cnp.NPY_ITER_MULTI_INDEX
```

Comment:
Is extracting this from the Python version worth it?

In either case, it's fine but maybe add short doc string that says that this is a limited implementation that recreates the `NpyIter` wrapped by a Python created `np.nditer` for testing?

(I started reading it as if it was some generic function, but it really is just a quick testing mock-up, which is fair!)

Reply:
it makes it easy to test the methods from cython if I extract what's already in the `nditer` object. It doesn't cover everything but allows some testing to be done easily. I added a docstring in https://github.com/numpy/numpy/pull/24691/commits/365b6ffc79493b8d18a5b30e4fff84e3859003ab